useragent/useragent_test.go

```diff
@@ -37,3 +37,11 @@ func TestUserAgent(t *testing.T) {
 	ua2 := UserAgent("FakeBeat", v, commit, buildTime, "integration_name/1.2.3")
 	assert.Regexp(t, regexp.MustCompile(`; integration_name\/1\.2\.3\)$`), ua2)
 }
+
+func TestUserAgentLite(t *testing.T) {
```

Comment:
Following the suggestion in https://github.com/elastic/elastic-agent-libs/pull/307/files#r2035942942, add a test to guarantee all permutations of the user agent under the 100 byte limit since we will have limited the set of arguments we can accept and can now test this explicitly.